nixos/modules/services/networking/opengfw.nix

```
@@ -0,0 +1,360 @@
+{ lib
+, pkgs
+, config
+, ...
+}:
+let
+  inherit (lib) mkOption types mkIf mdDoc optionalString;
+  cfg = config.services.opengfw;
+  format = pkgs.formats.yaml { };
+
+  settings =
+    if cfg.settings != null
+    then format.generate "opengfw-config.yaml" cfg.settings
+    else cfg.settingsFile;
+  rules =
+    if cfg.rules != [ ]
+    then format.generate "opengfw-rules.yaml" cfg.rules
+    else cfg.rulesFile;
+in
+{
+  options.services.opengfw = {
+    enable = lib.mkEnableOption (mdDoc "A flexible, easy-to-use, open source implementation of GFW on Linux.");
```

```suggestion
    enable = lib.mkEnableOption "OpenGFW, A flexible, easy-to-use, open source implementation of GFW on Linux";
```

`mdDoc` is no longer needed. Please remove them all. Also

https://github.com/NixOS/nixpkgs/blob/383dbcec8c6d03c35a91f93e9e90c7980b9de025/lib/options.nix#L91-L105